### ISSUE TITLE
 update documentation: VariableDeclarationUsageDistance on  block of initialization methods

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#13155]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
            <property name="allowedDistance" value="1"/>
        </module>
    </module>
</module>

/var/tmp $ cat Test.java
public class Test {
    public Test test(Object a) {
        Test obj = test("abc");
        int var = 12; // violation expected 'Dist.* between variable 'var'.* and its first usage is 4'
        obj.test("abc");
        obj.test("abc");
        obj.test("abc");
        int k = var + 3;
        return this;
    }

    Test fooFaaHooHaa() {
        int pc = 0; // violation expected, method chain is broken
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");

        fooFaaHooHaa().test("abc");

        test(pc).test("abc");
        return null;
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.2-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**
```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.x-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/vyom/IdeaProjects/Config-Checkstyle/src/Test.java:4:9: Distance between variable 'var' declaration and its first usage is 4, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
[ERROR] /home/vyom/IdeaProjects/Config-Checkstyle/src/Test.java:13:9: Distance between variable 'pc' declaration and its first usage is 8, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
Audit done.
Checkstyle ends with 2 errors.
```

There is unfinished work at https://github.com/checkstyle/checkstyle/pull/11672 and can be reused.

-------

Issue is fixed in some other update, no violations happening any more, but documentation should be updated to show example on how it works.

**Comment by Vyom-Yadav** [comment]
@romani Ping, created this in place in #11335

**Comment by Kevin222004** [comment]
I am on it

**Comment by Z4ID-Reaper** [comment]
@romani @Zopsss 
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

this is a genuine issue and should be included in google style label.
below i have used the same input files and config as used by the reporter of this issue.

---
### config.xml

```
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
            <property name="allowedDistance" value="1"/>
        </module>
    </module>
</module>
```
---
### Test.java
```
public class Test {
    public Test test(Object a) {
        Test obj = test("abc");
        int var = 12; // violation expected 'Dist.* between variable 'var'.* and its first usage is 4'
        obj.test("abc");
        obj.test("abc");
        obj.test("abc");
        int k = var + 3;
        return this;
    }

    Test fooFaaHooHaa() {
        int pc = 0; // violation expected, method chain is broken
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");
        test("abc").test("abc");

        fooFaaHooHaa().test("abc");

        test(pc).test("abc");
        return null;
    }
}
 
```
---
### cli output

```
D:\zaid\zaid\codes\asasas>java -jar checkstyle-10.23.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

**⚠ there should be error reported but it does not!**

**Comment by romani** [comment]
@Z4ID-Reaper , please use exact config from google style https://github.com/checkstyle/checkstyle/blob/52225c3a46dd3a7e71965c64c7a647b9696beff1/src/main/resources/google_checks.xml#L312 allowedDistance is not "1", default is "3"

**Comment by romani** [comment]
I do not believe it is good issue.

```
        int var = 12; // violation expected 'Dist.* between variable 'var'.* and its first usage is 4'
        obj.test("abc");
        obj.test("abc");
        obj.test("abc");
        int k = var + 3;
```

This Check is smart now to consider same block initialization of the same object `obj.xxxxx`  as single point for distance. 

see in documentation:
> Check can detect a block of initialization methods. If a variable is used in such a block and there are no other statements after variable declaration, then distance = 1. 

@Z4ID-Reaper , can you help us to improve documentation on `Check can detect a block of initialization methods. If a variable is used in such a block and there are no other statements after variable declaration, then distance = 1. ` looks like we compaletely missed to show config and real execution on such examples (there is codeblocks) but not a config and no violation comment to show behavior

without doc update it is very easy to be confused.

**Comment by Z4ID-Reaper** [comment]
Sure , i will make a PR soon with the required updates.
